### PR TITLE
Dialyzer work on test/example

### PR DESCRIPTION
**Pull request by klacke**
Found a few type errors in the quicer code as well. Especially, the cacert stuff must be part of listen opts for the server

### CODE REVIEW

**Comment by klacke** [comment]
I'm not quite sure what the rebar3 dialyzer call at the top Makefile actually does, but it's clearly not running dialyzer on all the code, especially not the test code.

**Comment by klacke** [comment]
To run dialyzer on the examples test code, I did, `make all` in `test/example` and then
`dialyzer -r .` at the top

File: test/example/qt.erl [hunk 1]
```diff
@@ -70,14 +70,14 @@ listener(L) ->
...
-  {ok, Conn} = quicer:accept(L, [], infinity),
+  {ok, Conn} = quicer:accept(L, #{}, infinity),
 
   Top ! {self(), more},
 
   io:format("accept -> ~p\n",[Conn]),
 
   case quicer:handshake(Conn) of
-    {ok, Conn} ->
+    {ok, _Conn} ->
```
Comment (qzhuyan) [inline test/example/qt.erl:80]: why this change?

Comment (klacke) [reply test/example/qt.erl:80]: Sigh, stupid, just  a misdirected attempt at getting through dialyzer, skip that

**Comment by qzhuyan** [comment]
I am not quiet sure if we need to run dialzyer over test code in the CI (at least not that urgent) but feel free to make a PR if you think it is necessary.

**Comment by klacke** [comment]
Thx.
As for dialyzer on the test code, since the test code is one of the primary sources for learning your lib, I feel it's important that the test code use the library the way it's intended to be used. I guess it's mostly an issue of the various options being a map as opposed to key,val list.
